client.go

```diff
@@ -1547,6 +1559,53 @@ func (c *ToznySDKV3) WriteFile(ctx context.Context, recordType string, plain map
 	return fileRecord, nil
 }
 
+type ReadFileOptions struct {
+	RecordID         string
+	DownloadFileName string
+}
+
+// ReadFile downloads and decrypts the file from the record
+func (c *ToznySDKV3) ReadFile(ctx context.Context, options ReadFileOptions) error {
+	secretUUID, err := uuid.Parse(options.RecordID)
+	if err != nil {
+		return err
+	}
+	fileResp, err := c.E3dbPDSClient.GetFileRecord(ctx, secretUUID)
+	if err != nil {
+		return err
+	}
+	fileURL := fileResp.Metadata.FileMeta.FileURL
+	downloadRequest := file.DownloadRequest{
+		URL:               fileURL,
+		EncryptedFileName: DefaultDownloadedFileName,
```

Comment:
This name should have some amount of entropy added to it, otherwise, this method is not thread-safe from the filesystems perspective, or will just block I am not positive on that off the top of my head. You could also change the DownloadFile method to use https://golang.org/pkg/io/ioutil/#example_TempFile which will generate a file that automatically has some entropy and claimes to guarantee no collisions (This is my recommendation but requires updates to e3db-clients-go which may dissuade). You could also just generate n bytes and base64 encode but then will need to make sure all characters are filesystem safe. Timestamps are another choice but I worry about the chance of collision if multiple go routines start at nearly the same time millisecond accuracy may not be enough given concern problem statement of running this in a go routine to download many files..

Reply:
This is potentially a problem with the DefaultEncryptedFileName for file upload as well. So if you do choose to do tempFiles I would recommend using temp files for upload as well.